### PR TITLE
Add support for `linguist-detectable` and `linguist-documentation`

### PR DESCRIPTION
**Pull request by KN4CK3R**
Add support for `linguist-detectable` and `linguist-documentation`
Add tests for the attributes

https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#detectable
https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#documentation

### CODE REVIEW

File: modules/git/repo_language_stats.go [hunk 1]
```diff
@@ -6,13 +6,35 @@ package git
...
+	"code.gitea.io/gitea/modules/optional"
 )
 
 const (
 	fileSizeLimit int64 = 16 * 1024   // 16 KiB
 	bigFileSize   int64 = 1024 * 1024 // 1 MiB
 )
 
+// true if "set"/"true", false if "unset"/"false", none otherwise
+func linguistToBool(attr map[string]string, name string) optional.Option[bool] {
```
Comment (silverwind) [inline modules/git/repo_language_stats.go]: Name it `gitAttributeToBool` and move it somewhere to the git module? Or does it only apply to those few attributes?

File: modules/git/repo_language_stats_gogit.go [hunk 2]
```diff
@@ -11,6 +11,7 @@ import (
 	"strings"
```
Comment (silverwind) [inline modules/git/repo_language_stats_gogit.go:11]: Seems like the touched gogit code is mostly duplicate. Move to a shared file?

File: modules/git/repo_language_stats.go (hunk 1 above)
Comment (KN4CK3R) [reply modules/git/repo_language_stats.go]: It's already in the git module but I can move it to some helper file.

File: modules/git/repo_language_stats_gogit.go (hunk 2 above)
Comment (KN4CK3R) [reply modules/git/repo_language_stats_gogit.go:11]: The differences are `continue / return nil`, `.Name() / .Name`, `.Size() / .Size` and content reading which makes it not as easy as it looks like. I would prefer to leave it as it is for this PR.

File: modules/git/repo_language_stats.go (hunk 1 above)
Comment (silverwind) [reply modules/git/repo_language_stats.go]: Ah yes, maybe `modules/git/attributes.go`.

File: modules/git/repo_language_stats_gogit.go (hunk 2 above)
Comment (silverwind) [reply modules/git/repo_language_stats_gogit.go:11]: Ok, should be refactored later.

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**